FSQLocationBroker/FSQLocationBroker.m

```diff
@@ -521,8 +521,8 @@ - (void)locationManager:(CLLocationManager *)manager didUpdateLocations:(NSArray
         }
     }
     
-    // only update currentLocation if mostRecentLocation happened after currentLocation
-    if ([self.currentLocation.timestamp laterDate:mostRecentLocation.timestamp] == mostRecentLocation.timestamp) {
+    // only update if currentLocation is not set or if mostRecentLocation happened after currentLocation
+    if (!self.currentLocation || [self.currentLocation.timestamp laterDate:mostRecentLocation.timestamp] == mostRecentLocation.timestamp) {
```

Comment:
I find the second half of that OR a little confusing... Do we have access to `NSDate+FS`'s `isAfterDate:` method here?